pkg/election/election_test.go

```diff
@@ -27,19 +29,51 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-var errTest = errors.New("testing")
+var (
+	errTest   = errors.New("testing")
+	errNoNode = fmt.Errorf("test error: %w", zk.ErrNoNode)
+)
+
+type Buffer struct {
+	b bytes.Buffer
+	m sync.Mutex
+}
+
+func (b *Buffer) Read(p []byte) (n int, err error) {
```

Comment:
I am wondering what are those functions for, but they are likely useful :)

Reply:
There's a hook to display logs when a unit test fails (helps with diagnosing concurrency), which has started triggering the race detector with the latest changes. I'll add a comment.